### DISCUSSION TITLE
USB Power Fault error on OpenFlops

### DISCUSSION THREAD

**Discussion by BK560A** [keirf/flashfloppy#730] [Q&A]
Hi all,
I'm using an OpenFlops flashed with Flash Floppy and I get a "USB Power Fault" when there is a USB drive attached.I know it means there is an overcurrent condition of the USB port.What could be the cause?I have other OpenFlops that do not throw this error.Thanks.

**Comment by keirf** [comment]
What microcontroller is used on this board? I don't think any openflops actually has a USB power switch, in which case this error condition ought to be impossible. Exactly when and how easily does this error occur?

**Reply by BK560A** [reply]
Thanks for reply.This is the OpenFlops I'm using :

https://github.com/SukkoPera/OpenFlops

It does not have a USB power switch.The error occurs after some seconds of operation.

P.S.

It has a STM32F105RBT6 MCU.

**Reply by keirf** [reply]
Firmware version?

**Reply by BK560A** [reply]
I'm using an old 3.18.

**Reply by keirf** [reply]
That's quite old. USB power fault is detected by low level at pin PA5. It might also be worth checking soldering around pin PA5 (which should be not connected to anything).

**Reply by BK560A** [reply]
Yes, I know it's quite old but it does his job so I don't feel the need to update it.OK, I will check for solder bridge.Thanks again for your support.

P.S.
Is it possible to generate a low signal pulse when an image is mounted?

**Reply by keirf** [reply]
Pins 2 and 34 are configurable and can carry disk change signal. You might want a different sort of signal though and on a different pin?

**Reply by BK560A** [reply]
Yes, I want an assertion when a disk image is mounted, from an unused pin of the MCU like 'PB13' (pin34).I have checked the source but I'm not a coder (I'm an hardware guy)

**Reply by BK560A** [reply]
Or better, more than an assertion I need a fixed low signal (0-0.8V) when a disk image is mounted.
